src/TestFramework/PhpUnit/CommandLine/ArgumentsAndOptionsBuilder.php

```diff
@@ -127,19 +128,22 @@ private function getMethodNameWithDataProvider(string $methodNameWithDataProvide
         $methodNameWithDataProviderResult = $methodNameWithDataProvider;
 
         /*
-         * in PHPUnit >=10 data providers with keys are stored as `Class\\test_method#some key`
-         * in PHPUnit <10 data providers with keys are stored as `Class\\test_method with data set "some key"`
+         * in PHPUnit >=10 data providers with keys are stored as `Class\\test_method#some key` or `Class\\test_method#0`
+         * in PHPUnit <10 data providers with keys are stored as `Class\\test_method with data set "some key"` or `Class\\test_method with data set #0`
          *
          * we need to translate to the old format because this is what PHPUnit <10 and >=10 understands from CLI `--filter` option
          */
         if (version_compare($testFrameworkVersion, '10', '>=')) {
             $methodNameParts = explode('#', $methodNameWithDataProviderResult, self::MAX_EXPLODE_PARTS);
 
             if (count($methodNameParts) > 1) {
-                $methodName = $methodNameParts[0];
-                $dataProviderKey = $methodNameParts[1];
+                [$methodName, $dataProviderKey] = $methodNameParts;
 
-                $methodNameWithDataProviderResult = sprintf('%s with data set "%s"', $methodName, $dataProviderKey);
+                if (preg_match('/^(\d+)$/', $dataProviderKey) === 0) {
```

Comment:
Isn't `is_numeric()` way faster than a regular expression? Also, the parentheses is redundant, isn't it?

Reply:
Here we go: #1904